Automattic/vip-go-mu-plugins
# Search: Update field mapping for meta types that are of DATE or DATETIME

## Description
Updates the subtree to include changes from https://github.com/Automattic/es-wp-query/pull/15

## Changelog Description

### Plugin Updated: Enterprise Search

When offloading meta queries that have a type of DATE or DATETIME, it will map the fields properly now.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Do a meta query with a type of date:
```
$args = array(
    'meta_query' => array(
        array(
            'key'     => 'meta_time',
            'value'   => '2009-01-01',
            'compare' => '<',
			'type' => 'DATE',
        ),
    ),
	'es' => true,
);
$query = new WP_Query( $args );
```
2) Go to Search Dev Tools, and see the below in the ES query:

```
        {
          "range": {
            "post_meta.meta_time.date": {
              "lt": "2009-01-01"
            }
          }
```

3) Apply PR and view ES query again and notice the difference in `post_meta.*` versus `meta.*`:

```
        {
          "range": {
            "meta.meta_time.date": {
              "lt": "2009-01-01"
            }
          }
```
